sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RequestHandler.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.restsvc.resources;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.stargate.sgv2.restsvc.models.RestServiceError;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Sgv2RequestHandler {
+  private static final Logger logger = LoggerFactory.getLogger(Sgv2RequestHandler.class);
+
+  public static Response handle(Callable<Response> action) {
+    try {
+      return action.call();
+    } catch (StatusRuntimeException grpcE) { // from gRPC
+      Status.Code sc = grpcE.getStatus().getCode();
+      switch (sc) {
+          // Start with codes we know how to handle
+        case INVALID_ARGUMENT:
+          return grpcResponse(
+              sc,
+              Response.Status.BAD_REQUEST,
+              "Invalid argument for gRPC operation",
+              grpcE.getMessage());
+        case NOT_FOUND:
+          return grpcResponse(
+              sc, Response.Status.NOT_FOUND, "Unknown gRPC operation", grpcE.getMessage());
+        case PERMISSION_DENIED:
+          return grpcResponse(
+              sc, Response.Status.UNAUTHORIZED, "Unauthorized gRPC operation", grpcE.getMessage());
+        case UNAUTHENTICATED:
+          return grpcResponse(
+              sc,
+              Response.Status.UNAUTHORIZED,
+              "Unauthenticated gRPC operation",
+              grpcE.getMessage());
+        case UNIMPLEMENTED:
+          return grpcResponse(
+              sc,
+              Response.Status.NOT_IMPLEMENTED,
+              "Unimplemented gRPC operation",
+              grpcE.getMessage());
+
+          // And then codes we ... don't know how to handle
```

Comment:
https://developers.google.com/maps-booking/reference/grpc-api/status_codes
https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md

This should help..

Reply:
Thank you!